crates/cubecl-runtime/src/memory_management/dynamic.rs

```diff
@@ -1,111 +1,207 @@
-use super::memory_pool::{
-    MemoryExtensionStrategy, MemoryPool, MemoryPoolBinding, MemoryPoolHandle, RoundingStrategy,
-    SmallMemoryPool,
+use super::{
+    memory_pool::{ExclusiveMemoryPool, MemoryPool, SliceBinding, SliceHandle, SlicedPool},
+    MemoryConfiguration, MemoryDeviceProperties, MemoryPoolOptions, MemoryUsage, PoolType,
 };
 use crate::storage::{ComputeStorage, StorageHandle, StorageId};
 use alloc::vec::Vec;
 
 use super::MemoryManagement;
 
-/// Reserves and keeps track of chunks of memory in the storage, and slices upon these chunks.
-pub struct DynamicMemoryManagement<Storage> {
-    min_chunk_alignment_offset: usize,
-    small_memory_pool: SmallMemoryPool,
-    pools: Vec<MemoryPool>,
-    options: Vec<MemoryPoolOptions>,
-    storage: Storage,
+enum DynamicPool {
+    Sliced(SlicedPool),
+    Simple(ExclusiveMemoryPool),
 }
 
-/// Options to initialize a [dynamic memory management](DynamicMemoryManagement).
-#[derive(new, Debug)]
-pub struct DynamicMemoryManagementOptions {
-    pools: Vec<MemoryPoolOptions>,
-    min_chunk_alignment_offset: usize,
-}
+// Bin sizes as per https://github.com/sebbbi/OffsetAllocator/blob/main/README.md
+// This guarantees that _for bins in use_, the wasted space is at most 12.5%. So as long
+// as bins have a high use rate this should be fairly efficient. That said, currently slices in
+// bins don't deallocate, so there is a chance more memory than needed is used.
+const EXP_BIN_SIZES: [usize; 200] = [
```

Comment:
Is there a function that can generate those bins? It would be cool to have it here as a comment!